Automattic/jetpack
# MyJetpack: redirect after checkout to Jetpack recommendations page if plugin is active

If we purchase a backup plan through Dashboard page in Jetpack plugin, it will redirect back to `admin.php?page=jetpack#/recommendations`, but if we purchase through My Jetpack, it will redirect back to My Jetpack.

We have a couple of issues created related to this interaction, involving Backup and Scan products. And after discussing in p6TEKc-6ZE-p2, we require to do the following changes in the interaction with Backup and Scan:

* If Jetpack plugin is active:
  * After purchasing a Backup or Scan plan, it should redirect to Jetpack recommendations page.
* If VaultPress Backup plugin is active:
  * After purchasing a Backup plan, it should redirect to backup plugin admin page.

Fixes #29506, fixes #29532 and fixes #29591.

## Proposed changes:

* Add new `post_checkout_url` field and `get_post_checkout_url` static method on MyJetpack Products class and implement it in Backup and Scan products to return the desired post checkout URL depending on Jetpack plugin availability.
* If Jetpack plugin is active, after purchasing Backup or Scan, customers will be redirected to `admin.php?page=jetpack#/recommendations`.
* If VaultPress Backup plugin is active and Jetpack is inactive, customers will be redirected to backup plugin admin page `admin.php?page=jetpack-backup`.

### Other information:


## Jetpack product discussion


* p1HpG7-l2l-p2
* p6TEKc-6ZE-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions - Backup

### Using Jetpack plugin
* Spin up a Jurassic Ninja site with Jetpack plugin
* Connect the site and user to Jetpack
* Navigate to Jetpack > My Jetpack
* Click on `Add VaultPress Backup` button under `Jetpack VaultPress Backup` product card
  <img src="https://user-images.githubusercontent.com/1488641/227659770-0186ed2d-1b30-4b2e-a775-69d43394dc1d.png" width="200" />
* Click on `Add Jetpack VaultPress Backup` button on the interstitial pricing page
  <img src="https://user-images.githubusercontent.com/1488641/227659840-7e9648ea-1284-40af-a51d-291a6a084fcd.png" width="200" />
* It should redirect you to the checkout page with the Jetpack VaultPress Backup 10 GB plan in the cart.
* Purchase the plan.
* It should redirect you back to `[site admin url]/admin.php?page=jetpack#/recommendations`. You should see this page:
  <img src="https://user-images.githubusercontent.com/1488641/227659958-4dc8b5c0-5d4f-44e9-a582-b0c672f740f9.png" width="200" />

#### Using Jetpack VaultPress Backup plugin and Jetpack plugin is inactive
* Spin up a Jurassic Ninja site with Jetpack VaultPress Backup plugin (Jetpack plugin should be inactive or not installed)
* Connect the site and user to Jetpack
* Navigate to Jetpack > My Jetpack
* Click on `Purchase` button under `Jetpack VaultPress Backup` product card
  <img src="https://user-images.githubusercontent.com/1488641/227660989-eacd8dd0-fada-47c3-a5a6-82ae1b3b19ef.png" width="200" />
* Click on `Add Jetpack VaultPress Backup` button on the interstitial pricing page
* It should redirect you to the checkout page with the Jetpack VaultPress Backup plan in the cart.
* Purchase the plan.
* Validate it should redirect you to the backup plugin admin page (/wp-admin/admin.php?page=jetpack-backup).

## Testing instructions - Scan
* Spin up a Jurassic Ninja site with Jetpack plugin
* Connect the site and user to Jetpack
* Navigate to Jetpack > My Jetpack
* Click on `Purchase` button under `Scan` product card
  <img src="https://user-images.githubusercontent.com/1488641/227668853-8f875f87-61e9-4c29-91e7-a3f821b68d25.png" width="200" />
* Click on `Add Jetpack Scan` button on the interstitial pricing page
* It should redirect you to the checkout page with the Jetpack Scan plan in the cart.
* Purchase the plan.
* It should redirect you back to `[site admin url]/admin.php?page=jetpack#/recommendations`. You should see this page:
  <img src="https://user-images.githubusercontent.com/1488641/227669334-56f99db0-02a9-4770-a416-de27941d4035.png" width="200" />
